### ISSUE TITLE
Usage/Syntax Question - How can I extract audio as mp3 and download video as mkv in the same run of youtube-dl?

### ISSUE THREAD

**Issue by kennbr34** [ytdl-org/youtube-dl#29292]
I am unsure if I'm trying to do something the program simply cannot do, or if I don't understand the syntax and usage properly.

I want to be able to grab a podcast link, download and embed the subs into the video as an mkv, and also extract the audio as an mp3. Ideally I'd like to do this with one run of youtube-dl but it doesn't seem possible.

When I run: 

`youtube-dl --write-sub --write-auto-sub --embed-subs --recode-video mkv --extract-audio --audio-format mp3 [youtube-link]`

It will:

1. Download the subtitles
2. Download the video
3. Download the mp3
4. Delete the video
5. Convert the mp3 to mkv (without the video)
6. Delete the original mp3
7. Embed the subtitles into the mkv
8. Delete the subtitles

So what I end up getting is a mkv file with only the mp3 audio and the subtitles embedded.

I saw from the program's info output that I could use _-k_ to prevent it from deleting files but this has two problems

1. The video file left is not converted to mkv
2. It leaves the subtitles for me to clean up myself

I thought maybe I could just leave the _--recode-video_ option off thinking the program would automatically merge the video and subtitles into mkv itself, and thus not try to convert the mp3 to mkv, but instead it:

1. Downloads the subtitles
2. Downloads the video
3. Downloads the audio to mp3
4. Deletes the video
5. Tries to merge the subtitles and audio
6. Tells me that it can only embed subtitles into mp4, webm or mkv files
7. Leaves the mp3 and subtitles

I haven't tried anything more, but I'm kind of assuming from all this that downloading the audio and video must be done separately, and with _--extract-audio_ it just assumes you only want to work with the audio file and disregard the video.

I figure I'll probably just have to end up using ffmpeg to extract the audio from the mkv after youtube-dl runs, but I'd really like to accomplish this with just the single run of youtube-dl.

**Comment by section83** [comment]
Yes, I doubt it can be done in one process.  It also depends on what formats are available – YT does vary them somewhat.  it would help if you can provide an example URL and a copy of the verbose log (if the log is very long, no need to paste here; just upload it).

**Comment by kikuyan** [comment]
How about `--exec "ffmpeg -y -loglevel repeat+warning -i file:{} -vn -acodec libmp3lame -q:a 5 file:{}.mp3"` instead of `--extract-audio --audio-format mp3`. The filename will not be ideal, though.

**Comment by kennbr34** [comment]
> Yes, I doubt it can be done in one process. It also depends on what formats are available – YT does vary them somewhat. it would help if you can provide an example URL and a copy of the verbose log (if the log is very long, no need to paste here; just upload it).

Okay, here's a quick example from Tom Segura's podcast

`youtube-dl -v --write-sub --write-auto-sub --embed-subs --recode-video mkv --extract-audio --audio-format mp3 https://www.youtube.com/watch?v=qjV7FmiltNU`

I've attached the log
[segura.log](https://github.com/ytdl-org/youtube-dl/files/6645158/segura.log)

> How about `--exec "ffmpeg -y -loglevel repeat+warning -i file:{} -vn -acodec libmp3lame -q:a 5 file:{}.mp3"` instead of `--extract-audio --audio-format mp3`. The filename will not be ideal, though.

That's a pretty good idea. I suppose I could just write a wrapper-script around the ffmpeg command that can also use sed to strip the extension off the filename.
